refactor: convert all mapper classes to extension functions

Noor-Serry opened the pull request:
# Pull Request Template

## Description
Provide a clear and concise description of what this pull request does. Include the purpose and context for the changes.
: convert all mapper classes to extension functions in repository component 
---

## Changes Made
List the changes introduced in this pull request:

- change all mapper classes 
- change in repositories


---

## Screenshots (if applicable)
Include screenshots or GIFs to showcase the changes, especially if they impact the UI.
the app is working correctly as previously 
<img width="720" height="1080" alt="Screenshot_20250804_172500" src="https://github.com/user-attachments/assets/9b3add6b-297f-40a9-a84e-9c198767545e" />

---

## Checklist
Please ensure the following tasks are completed:

- [x] My code follows the code style of this project
- [x] Changes have been tested manually and verified.
- [x] PR includes at most one single feature.

---

## Additional Comments
why I did that .
The reason I initially chose to use mappers as classes instead of extension functions was to make unit testing easier. However, I realized this approach negatively impacted the quality of the production code.
In some cases, like the examples shown down, the mapping logic requires passing additional arguments. While classes make this possible, they also reduce flexibility and result in bulky repository constructors filled with mapper dependencies.
<img width="764" height="423" alt="image" src="https://github.com/user-attachments/assets/1a037c62-dac5-4e06-a5e9-05e089817d3d" />
<img width="831" height="214" alt="image" src="https://github.com/user-attachments/assets/5cd0d168-6442-45a3-b72b-9859dae41d9a" />
<img width="788" height="501" alt="image" src="https://github.com/user-attachments/assets/d6a93cdb-802c-43c1-a7e3-7c995ed7652e" />

this was my mistake so that it was my responsibility to fix it

Noor-Serry committed:
refactor: convert all mapper classes to extension functions

nabilBouzineDev commented on repository/src/main/java/com/amsterdam/repository/mapper/shared/GenreMapper.kt:
```diff
@@ -3,49 +3,47 @@ package com.amsterdam.repository.mapper.shared
-        37L -> MovieGenre.WESTERN
-        else -> MovieGenre.ALL
-    }
+fun Long.toMovieGenre(): MovieGenre = when (this) {
```
Great Job Noor 👌, But, this should be a class since Long is a primitive type and shouldn't have this extension to an object.

Noor-Serry replied:
thanks nabil.
okay I will make it just function

eslam-99 commented on localDatasource/src/main/java/com/amsterdam/localdatasource/roomDataBase/datasource/MovieLocalDataSourceImpl.kt:
```diff
@@ -68,8 +68,7 @@ class MovieLocalDataSourceImpl @Inject constructor(
             PopularMovieDto(
                 movieId = movie.movieId,
-                storedLanguage = movie.storedLanguage,
-                dateAdded = movie.insertedDate
```
Changes we agreed on latest daily meeting is an extra task, not related to mappers

eslam-99 commented on repository/src/main/java/com/amsterdam/repository/mapper/local/TvShowGenreLocalMapper.kt:
Why deleting this, it's supposed to be used as `MovieWithCategoriesLocalMapper`

Noor-Serry replied:
All files have been replaced with extensions so that this file is no longer needed.
I've checked everything and it's working properly.

eslam-99 commented on repository/src/main/java/com/amsterdam/repository/mapper/shared/GenreMapper.kt:
```diff
@@ -3,49 +3,47 @@ package com.amsterdam.repository.mapper.shared
-        37L -> MovieGenre.WESTERN
-        else -> MovieGenre.ALL
-    }
+fun Long.toMovieGenre(): MovieGenre = when (this) {
```
Is it better to have extension on `Long`, or make it normal function?

nabilBouzineDev commented:
> this was my mistake so that it was my responsibility to fix it

Well said and well done @Noor-Serry ✅.

One more change can improve readibitlty is to name the function without post fixes f.e: toLocalDto, toRemoteDto, toEntity, and toUiState.

I think this will improve readability and  it'll be easier in the autocomplete. what do you think?

Noor-Serry committed:
refactor: standardize mapper function names to `toEntity` and `toLocalDto`

This commit refactors various mapper functions across the repository to use consistent naming conventions:

- Functions that map from DTOs (Data Transfer Objects) to domain entities are now named `toEntity`.
- Functions that map from remote DTOs to local DTOs are now named `toLocalDto`.
- List mapping functions are updated to `toEntityList` and `toLocalDtoList` accordingly.
- The `toSessionTypeEntity` function in `SessionTypeMapper.kt` has been renamed to `stringToSessionTypeEntity` for clarity.
- The `toTvShowGenre` and `toMovieGenre` functions in `GenreMapper.kt` are now top-level functions instead of extension functions.

These changes improve code readability and maintainability by establishing a clear and predictable naming pattern for mapping operations. Test files have also been updated to reflect these changes.

Noor-Serry committed:
Merge branch 'develop' of github.com:Amsterdam-Team/Aflami into refactor/convert-mapper-classes-to-extentions-functions-in-repositoy-component

# Conflicts:
#	repository/src/main/java/com/amsterdam/repository/dto/local/LocalMovieDto.kt
#	repository/src/main/java/com/amsterdam/repository/mapper/local/MovieLocalMapper.kt
#	repository/src/main/java/com/amsterdam/repository/mapper/local/MovieWithCategoriesLocalMapper.kt
#	repository/src/main/java/com/amsterdam/repository/mapper/local/TvShowLocalMapper.kt
#	repository/src/main/java/com/amsterdam/repository/mapper/local/TvShowWithCategoryLocalMapper.kt
#	repository/src/main/java/com/amsterdam/repository/mapper/remote/EpisodeRemoteMapper.kt
#	repository/src/main/java/com/amsterdam/repository/mapper/remote/MovieDetailRemoteMapper.kt
#	repository/src/main/java/com/amsterdam/repository/mapper/remote/MovieRemoteMapper.kt
#	repository/src/main/java/com/amsterdam/repository/mapper/remote/TvShowDetailsRemoteMapper.kt
#	repository/src/main/java/com/amsterdam/repository/mapper/remote/TvShowRemoteMapper.kt
#	repository/src/main/java/com/amsterdam/repository/mapper/remoteToLocal/MovieRemoteDetailsLocalMapper.kt
#	repository/src/main/java/com/amsterdam/repository/mapper/remoteToLocal/MovieRemoteLocalMapper.kt

Noor-Serry committed:
refactor: update mappers to remove unused fields and add video URL

This commit refactors several mappers:
- `MovieRemoteMapper`: Added `videoUrl` to the `Movie` entity.
- `MovieDetailRemoteMapper`: Populates the `videoUrl` in the `Movie` entity using the first available video.
- `TvShowLocalMapper` and `TvShowWithCategoryLocalMapper`: Removed the unused `productionCompanies` field.
- `TvShowRemoteMapper`: Removed the unused `productionCompanies` field.

Noor-Serry committed:
Merge branch 'develop' of github.com:Amsterdam-Team/Aflami into refactor/convert-mapper-classes-to-extentions-functions-in-repositoy-component

# Conflicts:
#	repository/src/main/java/com/amsterdam/repository/repository/AuthenticationRepositoryImpl.kt
